src/utils/socket.js

```diff
@@ -0,0 +1,50 @@
+const app = require("../app");
+const http = require("http");
+const socketIO = require("socket.io");
+
+const server = http.createServer(app);
+
+server.listen(process.env.PORT || 4000, () => {});
+
+const io = socketIO(server, {
+  cors: {
+    origin: "http://localhost:3000",
+  },
+});
+
+const lobbyNamespace = io.of("/lobby");
+lobbyNamespace.on("connection", (socket) => {
+  const currentUser = [];
+  const { user, userProfile } = socket.handshake.query;
+  currentUser.push({ user: user, userProfile: userProfile });
+
+  socket.on("updateChat", ({ user, content }) => {
+    socket.broadcast.emit("chatUpdated", { user, content });
+  });
+
+  socket.on("updateCurrentUser", () => {
+    socket.broadcast.emit("CurrentUserUpdated", currentUser);
+  });
+
+  socket.on("disconnect", () => {
+    const disconnectedUser = socket.handshake.query.user;
+    const index = currentUser.findIndex(
+      (userObj) => userObj.user === disconnectedUser,
+    );
+    if (index !== -1) {
+      currentUser.splice(index, 1);
+      io.emit("CurrentUserUpdated", currentUser);
+    }
+  });
+});
+
+const battlesNamespace = io.of(/^\/battles\/(.+)$/);
+battlesNamespace.on("connection", (socket) => {
+  const roomId = socket.nsp.name.split("/")[2];
+
+  socket.on("updateKey", ({ key }) => {
+    io.to(roomId).emit("keyUpdated", key);
+  });
```

Comment:
receive
```suggestion
  socket.on("send-key", ({ key, score }) => {
    io.to(roomId).emit("receive-key", key, score);
  });
```
key를 주고 받는 과정에서 canvas API에서 오차가 발생해 score도 현재 유저 화면에서 상대방이 현재 받은 점수와 오차가 생길 수도 있을 것 같아 점수또한 상대방 유저 화면에서 얻은 점수를 주고 받아 렌더링 해야할 것 같습니다!

Reply:
변경 사항을 추가 하겠습니다.